libbeat/idxmgmt/ilm/client_handler.go

```diff
@@ -205,6 +207,77 @@ func (h *esClientHandler) queryFeatures(to interface{}) (int, error) {
 	return status, nil
 }
 
-func (h *esClientHandler) access() ESClient {
-	return h.client
+// FileClientHandler implements the ClientHandler interface for writing to a file.
+type FileClientHandler struct {
+	client FileClient
+}
+
+// FileClient defines the minimal interface required for the ClientHandler to
+// prepare a policy and write alias.
+type FileClient interface {
+	GetVersion() common.Version
```

Comment:
do we need the GetVersion here? Would it make sense for the version to be a parameter to NewFileClientHandler?

Reply:
The `GetVersion` enables to run some of the same version/ilm enabled checks for the `FileClient` as we run for the `ESClient. The idea behind this is that users would see the same result when they export ilm-policies and templates to stdout or a file, as when they finally run the setup cmd, using the ES client. Removing `GetVersion` from the interface would not allow for the version checks anymore.